src/modules/peek/Peek.FilePreviewer/Previewers/ImagePreviewer/Helpers/WICHelper.cs

```diff
@@ -14,15 +16,29 @@ public static Task<Windows.Foundation.Size> GetImageSize(string filePath)
         {
             return Task.Run(() =>
             {
-                // TODO: Find a way to get file metadata without hydrating files. Look into Shell API/Windows Property System, e.g., IPropertyStore
-                IWICImagingFactory factory = (IWICImagingFactory)new WICImagingFactoryClass();
-                var decoder = factory.CreateDecoderFromFilename(filePath, IntPtr.Zero, StreamAccessMode.GENERIC_READ, WICDecodeOptions.WICDecodeMetadataCacheOnLoad);
-                var frame = decoder?.GetFrame(0);
                 int width = 0;
                 int height = 0;
 
-                // TODO: Respect EXIF data and find correct orientation
-                frame?.GetSize(out width, out height);
+                PropertyStoreShellApi.IPropertyStore? propertyStore = null;
```

Comment:
I would recommend moving this out of the wichelper as the propertystore is not related to WIC.

Reply:
You could create a PropertyHelper with like PropertyHelper.GetWidth(filename), PropertyHelper.GetHeight(filename).

Reply:
Then have the logic between either propertstore and wic in the ImagePreviewer.GetPreviewSizeAsync().

Reply:
Thanks for the suggestion.  I just made a GetImageSize for now since Width and Height aren't needed independently yet.  Let me know what you think.